### ISSUE TITLE
Long text messages get clipped and unseeable at the bottom

### ISSUE THREAD

**Issue by gsbabil** [signalapp/Signal-iOS#3032]
- [X] I have searched open and closed issues for duplicates

----------------------------------------

### Bug description

- Long texts sent in a single message, particularly the ones sent from
  Signal Desktop are clipped at the bottom in Signal-iOS

- There is a "Tap for More" button at the bottom of the long text,
  but screen this button leads to also does not seem to show the
  full message

### Steps to reproduce

- Send a long message, e.g. 48 lines, 1477 characters, to a contact
  from Signal-Desktop

- Open the same sent-message from Signal-iOS

- The message will be shown as clipped at the bottom with a "Tap for
  More" button at the bottom

**Actual result:**
  
- The long message is clipped at the bottom at around line 17,
  character 573

**Expected result:** 

- Full message should be displayed at least with "Tap for More"

### Device info

- **Device**         : `iPhone 6`
- **iOS version**    : `11.2.5 (15D60)`
- **Signal version** : `2.19.7.6`

### Link to debug log

- https://gist.github.com/7729ff024f281f3b63a4493696e41bfb

### Screenshots

![img_2018](https://user-images.githubusercontent.com/1061108/36360964-82bce73a-157c-11e8-9a60-dcfce1ef673c.PNG)

**Comment by michaelkirk** [comment]
We intentionally truncate very long messages in the conversation view, hence the "Tap For More" button.

> There is a "Tap for More" button at the bottom of the long text,
but screen this button leads to also does not seem to show the
full message

You say... "does not **seem**". Can you please verify the message was truncated?

You said you had a linked desktop, do you see different text for this message on desktop than you do on the screen you're shown after tapping "Tap For More" on iOS?

If you can share a message that reproduces this truncated behavior, that would be helpful.

**Comment by gsbabil** [comment]
Hi @michaelkirk,

> You say... "does not seem". Can you please verify the message was
> truncated?

Yes.

If I tap on "Tap on More" in the iOS app, the next screen still does
_not_ show the full message.

> You said you had a linked desktop, do you see different text for
> this message on desktop than you do on the screen you're shown
> after tapping "Tap For More" on iOS?

No.

The Signal-Desktop shows the full message without truncation.

> If you can share a message that reproduces this truncated
> behavior, that would be helpful.

Please find a redacted version of the message below.

Thank you.


```
xx xxxxxx,

xxxxxx xxxx x xxxx xx xxx xxxxx x xxxx xxxx xxx xxxxxxxx xxxxxx xxxxx xxx.


xxxxxxxxxxxxxxxxxxxxxxxxxx
xx xxxxxxx

x x xxxxx xxxxxx xxxx xxxxxx xxxx xxxxx xx xxxxxxx xxxx xxx xxxxxx x xxxxx xx xxxxxx xx xxxx xxx xxx

xxxxx xxxx

x xxxx xxxx xxx x xxxxxx xxxxxxxx xxxxxx xxx xxxx xxxxxxx xxx xxx xxxxx. xxxxxx xxxxxxxxxxxx xx xxx xxxxx xxxx xxxxxxx xxxxxxxxx xx xxxx xxxx

xxx xxxxxxxx xxxxxx xxx xxx xxxxxxxxx xxxxxxxxxxx xxxxxxxx. xxxxx xxx xx xxxxxxxxxx xxxxx xx xxxxxxx xxx xxx xxxx xx xxxxxx xxxxxx xxx xxxxx xxxx xxxxxx xx xxxxxx. xxx xxxx xxxxxxx xxx x xxxx xx xxxx xxxxx xx xxxxxx xx xxxx xxx xxx xxxx xx xxxxxx xxxxxxxx

xxxxx xx xxx xxx xxxx xxx xxx xxxx xxxx xxxxx xxx xxx xxxxxxxxxxxx xx xxxxxxx xx xx xxxxxxxxxxxxx xxx xxxxxxxxxxx xxxxx xxxxxx xxxx xxxxx xxx xxxxxxx xx xxxx

xxx xxxxx xx xxxxx xx xxxx xxxxxxx xx xxxxxxxx xxx xxxx xxxxxx xx xx xxxxxxx xxx xxxx xx xx xx xxxx xxxxx xxx xxxxxx xxxxx xxxxxxxxxx.  x xxxx xxxx xxxx xxx xxxxxx x xxxxxxxx xxxxxxxxxxx

xxxxxxx xxxxxxx xx xxxx xx xxx xxxxx xxxx xxxxx xxxxxxx


xx xxxxxxx xxxxxxx

  xx xxxxx

      xxxxxx x xxxxxxxxxx
      xxxxx  x xxxxxxx

  xx xxxxxxx

      x xxxx xxxxxxxx xx xxxx xxxxxxxx xxxxxxx

  xx xxxxxxx

      x xxxxxx x xxxxxxxxxxxxx
      x xxxxx  x xxxxxxxxxxxx

      x xxxxxx xxxxxx xxxxxxx

          xxxx xxxx xxxx xxxx
          xxxx xxxx xxxx xxxx
          xxxx xxxx xxxx xxxx
          xxxx xxxx xxxx xxxx
          xxxx xxxx xxxx xxxx
          xxxx xxxx xxxx xxxx

xxxxxxxxxxxxxxxxxxxxxxxx
```

**Comment by JoshuaaMichael** [comment]
Confirmed this is also an issue for me. I was coming to report this but found this open.

When I send a long message through the desktop app, it is shrunk with a "tap for more" in the iOS app. You've stated above this is normal behavior, that's fine. 

However when I tap to read more, it opens a new page, but doesn't show the rest of the message. 

I think the page renders the message onto the screen and into the textbox, however it's the textbox field that doesn't extend further for the long message. I believe this because when I scroll down the page, I scroll as if the whole long message is actually there (I can see the scroll bar on the side scroll down), however it's not moving the message shown on screen at all, it's static while it scrolls down the page.

It's worth noting that strangely when I opened and close that page a few times it'll give me one or two more lines of the message, but no more than that, I'm still missing a lot of my message.

The message I sent was 31 lines long. It was 284 words, and 1454 characters including spaces (1164 without).

Hope this makes sense, feel free to ask for clarification if any is needed.

Regards,
